### PR TITLE
Disallow firing staff who are currently fixing or inspecting rides

### PR DESCRIPTION
**Pull request by ZehMatt**
This may lock up the ride when staff is fired at the right time, since staff manipulates the state of the ride status there is no work-around without enabling exploits that people can use to immediately fix rides by sending mechanics in and then immediately fire them. I think this solution is the right way to solve the problem without adding any complex extra logic on top of.

Here is a park where I isolated the problem:
[0_Everything_Park_Unfixable_ride.zip](https://github.com/OpenRCT2/OpenRCT2/files/11523181/0_Everything_Park_Unfixable_ride.zip)

With this PR when you try to fire mechanics that are either inspecting or fixing the ride you'll get an error as shown below:
openrct2_2023-05-20_23-39-509be846cea41947e06

Also in order to reproduce this you have to fire the mechanic at the right time, ActionFrame has to be 0x28 in order for this bug to trigger, that's where it sets the new mechanic status on the ride and would set it to completed in another sub state.

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Would it be easy / feasible to disable the fire button in the interface as well, depending on the staff/mechanic status?

**Comment by karst** [comment]
> Would it be easy / feasible to disable the fire button in the interface as well, depending on the staff/mechanic status?

An issue then would be that if someone initiated the fire button before entering it will still have the pop-up so you could still actually get rid of them. Tho greying out the bin might be a clean extra.

**Comment by AaronVanGeffen** [comment]
> An issue then would be that if someone initiated the fire button before entering it will still have the pop-up so you could still actually get rid of them. Tho greying out the bin might be a clean extra.

Yes, I realised the game action would still need to perform the check. Greying out the button is just for UX purposes.

**Comment by ZehMatt** [comment]
> Would it be easy / feasible to disable the fire button in the interface as well, depending on the staff/mechanic status?

Not a bad idea, however might be best to separate the UI change from this logical change.

File: data/language/en-GB.txt [hunk 1]
```diff
@@ -3693,6 +3693,8 @@ STR_6587    :The OpenRCT2 Title Theme is a work of Allister Brimble,{NEWLINE}lic
 STR_6588    :Thanks to Herman Riddering for allowing us to record the 35er Voigt.
 STR_6589    :Show window buttons on the left
 STR_6590    :Show the window buttons (e.g. to close the window) on the left of the title bar instead of on the right.
+STR_6591    :Staff is currently fixing a ride and can't be fired.
```
Comment (rik-smeets) [inline data/language/en-GB.txt:3696]:
```suggestion
STR_6591    :Staff member is currently fixing a ride and can't be fired.
```

File: data/language/en-GB.txt [hunk 2]
```diff
@@ -3693,6 +3693,8 @@ STR_6587    :The OpenRCT2 Title Theme is a work of Allister Brimble,{NEWLINE}lic
...
+STR_6592    :Staff is currently inspecting a ride and can't be fired.
```
Comment (rik-smeets) [inline data/language/en-GB.txt:3697]:
```suggestion
STR_6592    :Staff member is currently inspecting a ride and can't be fired.
```

**Review by rik-smeets — COMMENTED** [review]
LGTM, just have two suggestions for the string literals.

**Comment by rik-smeets** [comment]
I also added the disabling of the fire button in the window in a separate commit.

**Review by rik-smeets — APPROVED** [review]

File: data/language/en-GB.txt [hunk 3]
```diff
@@ -3693,6 +3693,8 @@ STR_6587    :The OpenRCT2 Title Theme is a work of Allister Brimble,{NEWLINE}lic
 STR_6588    :Thanks to Herman Riddering for allowing us to record the 35er Voigt.
 STR_6589    :Show window buttons on the left
 STR_6590    :Show the window buttons (e.g. to close the window) on the left of the title bar instead of on the right.
+STR_6591    :Staff member is currently fixing a ride and can't be fired.
```
Comment (ZehMatt) [inline data/language/en-GB.txt:3696]:
```suggestion
STR_6591    :Staff member is currently fixing a ride and can’t be fired.
```

**Comment by ZehMatt** [comment]
Tested and all good, unfortunately people who ended up in this situation have to use cheats to make the ride work again, I don't want to make any assumptions during import about a complicated state especially when staff is the one where the sub state drives the repair/inspection state, I rather not make any assumptions, its even possible that a new mechanic has been hired meanwhile who has the same id as the one who previously was on the ride, so from here on its impossible to get into this dilema.

**Pull request merged**